src/org/jitsi/impl/neomedia/rtp/RTPEncodingImpl.java

```diff
@@ -279,11 +279,15 @@ public boolean matches(byte[] buf, int off, int len)
             return false;
         }
 
+        if (temporalId == -1)
+        {
+            return true;
+        }
+
         int tid = track.getMediaStreamTrackReceiver()
             .getStream().getTemporalID(buf, off, len);
 
-        return (tid == -1 && ArrayUtils.isNullOrEmpty(dependencyEncodings))
-                || tid == temporalId;
+        return tid == temporalId;
```

Comment:
OK, I don't mind merging it this way. But I think it's pretty clear how this works -- you send 3 independent simulcast streams, and each stream is encoded without temporal layers. And that's a completely valid usage. Simulcast and TS are completely separate, and we are superficially adding an extra assumption that simulcast streams must always include the tid field.